processor/src/decoder/tests.rs

```diff
@@ -1056,6 +1049,31 @@ fn call_block() {
         assert_eq!(sys_trace[FMP_COL_IDX][i], FMP_MIN + TWO);
     }
 
+    // --- check the in_syscall column ------------------------------------------------------------
+
+    // since no syscalls were made, values in the syscall flag column should be all ZEROs
+    assert_eq!(
+        &sys_trace[IN_SYSCALL_COL_IDX][..trace_len],
+        vec![ZERO; trace_len]
+    );
+
+    // --- check fn hash columns ------------------------------------------------------------------
+
+    // before the CALL operation is executed, we are in a root context and thus fn_hash is ZEROs.
+    for i in 0..8 {
+        assert_eq!(get_fn_hash(&sys_trace, i), [ZERO; 4]);
+    }
+
```

Comment:
In general, I agree - though I think it would be difficult to split up in this case without duplicating most of the test logic. I might be wrong though - so, definitely worth revisiting in the future.

Reply:
I agree with you as far as trying to do decoder and system unit tests goes, but I don't see why it would need to be split up if we lifted this test out of the decoder to the processor level.